src/scheduled_tasks.rs

```diff
@@ -356,48 +357,156 @@ async fn overwrite_deleted_posts_and_comments(pool: &mut DbPool<'_>) {
+  #[diesel(sql_type = Integer)]
+  community_id: i32,
+}
+
+/// Re-calculate the site and community active counts for a given interval
+async fn active_counts(pool: &mut DbPool<'_>, interval: (&str, &str)) {
+  info!(
+    "Updating active site and community aggregates for {}...",
+    interval.0
+  );
 
   let conn = get_conn(pool).await;
 
   match conn {
     Ok(mut conn) => {
-      let intervals = vec![
-        ("1 day", "day"),
-        ("1 week", "week"),
-        ("1 month", "month"),
-        ("6 months", "half_year"),
-      ];
-
-      for i in &intervals {
-        let update_site_stmt = format!(
-      "update site_aggregates set users_active_{} = (select * from r.site_aggregates_activity('{}')) where site_id = 1",
-      i.1, i.0
-    );
-        sql_query(update_site_stmt)
-          .execute(&mut conn)
-          .await
-          .map_err(|e| error!("Failed to update site stats: {e}"))
-          .ok();
-
-        let update_community_stmt = format!("update community_aggregates ca set users_active_{} = mv.count_ from r.community_aggregates_activity('{}') mv where ca.community_id = mv.community_id_", i.1, i.0);
-        sql_query(update_community_stmt)
-          .execute(&mut conn)
-          .await
-          .map_err(|e| error!("Failed to update community stats: {e}"))
-          .ok();
-      }
-
-      info!("Done.");
+      process_site_aggregates(&mut conn, interval).await;
+      process_community_aggregates(&mut conn, interval).await;
     }
     Err(e) => {
       error!("Failed to get connection from pool: {e}");
     }
   }
 }
 
+/// Re-calculate all the active counts
+async fn all_active_counts(pool: &mut DbPool<'_>, intervals: [(&str, &str); 4]) {
+  for i in intervals {
+    active_counts(pool, i).await;
+  }
+}
+
+async fn process_site_aggregates(conn: &mut AsyncPgConnection, interval: (&str, &str)) {
+  // Select the site count result first
+  let site_activity_res = sql_query(format!(
+    "select * from r.site_aggregates_activity('{}')",
+    interval.0
+  ))
+  .get_result::<SiteActivitySelectResult>(conn)
+  .await;
+
+  match site_activity_res {
+    Ok(res) => {
+      let processed_rows = res.site_aggregates_activity;
+
+      // Update the site count
+      sql_query(format!(
+        "update site_aggregates set users_active_{} = $1 where site_id = 1",
+        interval.1,
+      ))
+      .bind::<Integer, _>(processed_rows)
+      .execute(conn)
+      .await
+      .map_err(|e| error!("Failed to update site stats: {e}"))
+      .ok();
+
+      info!(
+        "Finished site_aggregates active_{} (processed {} rows)",
+        interval.1, processed_rows
+      );
+    }
+    Err(e) => {
+      error!("Failed to fetch site activity: {e}");
+    }
+  }
+}
+
+async fn process_community_aggregates(conn: &mut AsyncPgConnection, interval: (&str, &str)) {
+  // Select the community count results into a temp table.
+  let caggs_temp_table = "community_aggregates_temp_table";
```

Comment:
Include the interval here, just in case there are multiple calls to this function at the same time.

Reply:
Fixed.